oteapi_dlite/strategies/oceanlab_influx_parser.py

```diff
@@ -0,0 +1,253 @@
+
+        try:
+            env = jinja2.Environment(loader=jinja2.BaseLoader, autoescape=True)
+            env.globals.update(enumerate=enumerate, str=str)
+            bucket = f"{config.DATABASE}/{config.RETPOLICY}"
+            configuration = {
+                "bucket": bucket,
+                "timeRange": "-12h",
+                "limitSize": "50",
+                "measurements": [
+                    {
+                        "measurement": "ctd_conductivity_munkholmen",
+                        "field": "conductivity",
+                    },
+                    {
+                        "measurement": "ctd_density_munkholmen",
+                        "field": "density",
+                    },
+                    {
+                        "measurement": "ctd_salinity_munkholmen",
+                        "field": "salinity",
+                    },
+                    {
+                        "measurement": "ctd_pressure_munkholmen",
+                        "field": "pressure",
+                    },
+                ],
+            }
+            tmpl = env.from_string(TEMPLATE)
+            flux_query = tmpl.render(configuration).strip()
+            columns = query_to_df(
+                flux_query, config.url, config.USER, config.PASSWORD
+            )
+        except Exception as e:
+            # Handle errors that occur during JSON parser instantiation or
+            # data retrieval. You can log the exception, raise a custom
+            # exception, or handle it as needed. For example, logging the
+            # error and raising a custom exception:
+            print(f"Error during JSON parsing: {e}")
+            raise RuntimeError("Failed to parse JSON data.") from e
+
+        # Create DLite instance
+        meta = get_meta(self.parse_config.entity)
+        inst = meta(dims=[configuration["limitSize"]])
+
+        for name in [
+            measurement["field"]  # type: ignore
+            for measurement in configuration["measurements"]
+        ]:
+            inst[name] = columns[name]
+        inst["time"] = [
+            d.strftime("%m/%d/%Y, %H:%M:%S") for d in columns["_time"]
+        ]
+        # Add collection and add the entity instance
+        coll = get_collection(
+            collection_id=self.parse_config.configuration.collection_id
+        )
+        print(coll)
+        coll.add(config.label, inst)
+        update_collection(coll)
+
+        return InfluxParseSessionUpdate(
+            collection_id=coll.uuid,
+            inst_uuid=inst.uuid,
+            label=config.label,
+        )
+
+
+@cachetools.cached(cache=cachetools.LRUCache(maxsize=128))
+def query_to_df(query, url, USER, PASSWORD):
+    """query_to_df"""
+    with influxdb_client.InfluxDBClient(
+        url=url, token=f"{USER}:{PASSWORD}"
+    ) as client:
+        df = client.query_api().query_data_frame(query)
+    return df
+
+
+# Define the Jinja2 template
```

Comment:
Please try to add more semantic information in this comment, like that the template is for an InfluxDB query, maybe add what kind of query, what kind of things does it do? If you're not completely sure that's fine (but also sort of the point of having such a comment). I, at least, have not clue of what's going on in this query string 😅